src/qibolab/designs/qmrs.py

```diff
@@ -0,0 +1,160 @@
+from qibo.config import log, raise_error
+
+from qibolab.designs.abstract import AbstractInstrumentDesign
+from qibolab.platforms.utils import Channel
+
+
+class QMRSDesign(AbstractInstrumentDesign):
+    """Instrument design for Quantum Machines (QM) OPXs and Rohde Schwarz local oscillators.
+
+    IPs and other instrument related parameters are hardcoded in ``__init__`` and ``setup``.
+
+    Args:
+        address (str): Address and port for the QM OPX cluster.
+            Default is the current address for the instruments in TII lab.
+        simulation_duration (int): Duration for the simulation in ns.
+            If given the compiler simulator will be used instead of the actual hardware.
+            Default is ``None`` which falls back to the hardware.
+        cloud (bool): See :class:`qibolab.instruments.qmsim.QMSim` for details.
+            Relevant only when ``simulation_duration`` is given.
+
+    Attributes:
+        is_connected (bool): Boolean that shows whether instruments are connected.
+        opx (:class:`qibolab.instruments.qm.QMOPX`): Object used for controlling the QM OPXs.
+        local_oscillators (list): List of local oscillator objects.
+            instrument objects.
+    """
+
+    def __init__(self, address="192.168.0.1:80", simulation_duration=None, cloud=False):
+        from qibolab.instruments.rohde_schwarz import SGS100A
+
+        self.is_connected = False
+
+        # Instantiate QM OPX instruments
+        if simulation_duration is None:
+            from qibolab.instruments.qm import QMOPX
+
+            self.opx = QMOPX("qmopx", address)
+        else:
+            from qibolab.instruments.qmsim import QMSim
+
+            self.opx = QMSim("qmopx", address, simulation_duration, cloud)
+
+        # Map controllers to qubit channels (HARDCODED)
+        # readout
+        Channel("L3-25_a").ports = [
+            ("con1", 10),
+            ("con1", 9),
+        ]
+        Channel("L3-25_b").ports = [
+            ("con2", 10),
+            ("con2", 9),
+        ]
+        # feedback
+        Channel("L2-5").ports = [("con1", 1), ("con1", 2)]
+        # drive
+        Channel("L3-11").ports = [("con1", 1), ("con1", 2)]
+        Channel("L3-12").ports = [("con1", 3), ("con1", 4)]
+        Channel("L3-13").ports = [("con1", 5), ("con1", 6)]
+        Channel("L3-14").ports = [("con1", 7), ("con1", 8)]
+        Channel("L3-15").ports = [("con3", 1), ("con3", 2)]
+        # flux
+        Channel("L4-1").ports = [("con2", 1)]
+        Channel("L4-2").ports = [("con2", 2)]
+        Channel("L4-3").ports = [("con2", 3)]
+        Channel("L4-4").ports = [("con2", 4)]
+        Channel("L4-5").ports = [("con2", 5)]
```

Comment:
Same as above, we should decide whether this information will be stored in the code, like I do here, or is imported from another file. If we keep it in the code, sure I can replace `"con"` with integers. This will work for QM but may not work for other platforms/designs.